frontends/web/src/routes/device/bitbox02/backups.tsx

```diff
@@ -42,132 +37,116 @@ interface BackupsProps {
+    restoreBackup(deviceID, selectedBackup)
       .then(({ success }) => {
-        this.setState({
-          restoring: false,
-          errorText: success ? '' : this.props.t('backup.restore.error.general'),
-        });
-        if (this.props.onRestoreBackup) {
-          this.props.onRestoreBackup(success);
+        setRestoring(false);
+        setErrorText(success ? '' : t('backup.restore.error.general'));
+        if (onRestoreBackup) {
+          onRestoreBackup(success);
         }
       });
   };
 
-  public render() {
-    const {
-      t,
-      children,
-      backups,
-      showRestore,
-      showCreate,
-      showRadio,
-      deviceID,
-    } = this.props;
-    const { selectedBackup,
-      restoring,
-      errorText,
-    } = this.state;
-    if (!backups.success) {
-      return <div>Error fetching backups</div>;
-    }
+  if (backups === undefined) {
     return (
-      <div>
-        <div className={backupStyle.stepContext}>
+      <VerticallyCenteredSpinner />
+    );
+  }
+
+  if (!backups.success) {
+    return <div>Error fetching backups</div>;
+  }
+
+  return (
+    <div>
+      <div className={backupStyle.stepContext}>
+        {
+          errorText && (
+            <Toast theme="warning">
+              {errorText}
+            </Toast>
+          )
+        }
+        <div className={backupStyle.backupsList}>
+          {
+            backups.backups!.length ? (
+              <div className={backupStyle.listContainer}>
+                {
+                  backups.backups!.map(backup => (
+                    <div key={backup.id} className={backupStyle.item}>
+                      <BackupsListItem
+                        disabled={restoring}
+                        backup={backup}
+                        selectedBackup={selectedBackup}
+                        handleChange={(b => setSelectedBackup(b))}
+                        onFocus={() => undefined}
+                        radio={showRadio} />
+                    </div>
+                  ))
+                }
+              </div>
+            ) : (
+              <p>{t('backup.noBackups')}</p>
+            )
+          }
+        </div>
+        <div className={backupStyle.backupButtons}>
+          {
+            showRestore && (
+              <Button
+                primary={true}
+                disabled={!selectedBackup || restoring}
+                onClick={restore}>
+                {t('button.restore')}
+              </Button>
+            )
+          }
+          {
+            showCreate && (
+              <Create deviceID={deviceID} />
+            )
+          }
           {
-            errorText && (
-              <Toast theme="warning">
-                {errorText}
-              </Toast>
+            showCreate && (
+              <Check
+                deviceID={deviceID}
+                backups={backups.backups ? backups.backups : []}
+                disabled={backups.backups!.length === 0}
```

Comment:
There is a few exlamantion marks which can be removed if the type explicitly only passes backups in the success response.

```diff
diff --git a/frontends/web/src/api/bitbox02.ts b/frontends/web/src/api/bitbox02.ts
index bed0432fc..a523dbc55 100644
--- a/frontends/web/src/api/bitbox02.ts
+++ b/frontends/web/src/api/bitbox02.ts
@@ -137,19 +137,19 @@ export const restoreFromMnemonic = (
   return apiPost(`devices/bitbox02/${deviceID}/restore-from-mnemonic`);
 };
 
-export type SubscribedBackup = {
-    success: boolean;
-    backups?: Backup[];
+export type BackupResponse = {
+    success: true;
+    backups: Backup[];
 }
 
 export const getBackupList = (
   deviceID: string
-): Promise<SubscribedBackup> => {
+): Promise<FailResponse | BackupResponse> => {
   return apiGet(`devices/bitbox02/${deviceID}/backups/list`);
 };
 
 export const subscribeBackupList = (deviceID: string) => (
-  (cb: TSubscriptionCallback<SubscribedBackup>) => (
+  (cb: TSubscriptionCallback<FailResponse | BackupResponse>) => (
     subscribeEndpoint(`devices/bitbox02/${deviceID}/backups/list`, cb)
   )
 );
diff --git a/frontends/web/src/routes/device/bitbox02/backups.tsx b/frontends/web/src/routes/device/bitbox02/backups.tsx
index b97619a8f..66a5f1d54 100644
--- a/frontends/web/src/routes/device/bitbox02/backups.tsx
+++ b/frontends/web/src/routes/device/bitbox02/backups.tsx
@@ -57,7 +57,7 @@ export const BackupsV2 = ({
     if (!backups) {
       return;
     }
-    if (!selectedBackup || !backups.backups) {
+    if (!selectedBackup || !backups.success) {
       return;
     }
     const backup = backups.backups.find(b => b.id === selectedBackup);
@@ -98,10 +98,10 @@ export const BackupsV2 = ({
         }
         <div className={backupStyle.backupsList}>
           {
-            backups.backups!.length ? (
+            backups.backups.length ? (
               <div className={backupStyle.listContainer}>
                 {
-                  backups.backups!.map(backup => (
+                  backups.backups.map(backup => (
                     <div key={backup.id} className={backupStyle.item}>
                       <BackupsListItem
                         disabled={restoring}
@@ -140,7 +140,7 @@ export const BackupsV2 = ({
               <Check
                 deviceID={deviceID}
                 backups={backups.backups ? backups.backups : []}
-                disabled={backups.backups!.length === 0}
+                disabled={backups.backups.length === 0}
               />
             )
           }
```